distributed/shuffle/tests/test_shuffle.py

```diff
@@ -0,0 +1,197 @@
+import asyncio
+import io
+import shutil
+from collections import defaultdict
+
+import pandas as pd
+import pytest
+
+pa = pytest.importorskip("pyarrow")
+
+import dask
+import dask.dataframe as dd
+
+from distributed.shuffle.shuffle_extension import (
+    dump_batch,
+    list_of_buffers_to_table,
+    load_arrow,
+    split_by_partition,
+    split_by_worker,
+)
+from distributed.utils_test import gen_cluster
+
+
+@gen_cluster(client=True, timeout=1000000)
+async def test_basic(c, s, a, b):
+    df = dask.datasets.timeseries(
+        start="2000-01-01",
+        end="2000-01-10",
+        dtypes={"x": float, "y": float},
```

Comment:
I would suggest to remove this constraint and just use the ordinary timeseries. If we include the `name` column, this covers at least the case where we're casting the meta column to string